mythril/laser/ethereum/instructions.py

```diff
@@ -953,79 +942,181 @@ def call_(self, global_state):
+            callee_address, callee_account, call_data, _, call_data_type, gas, _, _ = get_call_parameters(global_state,
+                                                                                                          self.dynamic_loader)
         except ValueError as e:
             logging.info(
                 "Could not determine required parameters for call, putting fresh symbol on the stack. \n{}".format(e)
             )
             global_state.mstate.stack.append(BitVec("retval_" + str(instr['address']), 256))
             return [global_state]
 
-        global_state.mstate.stack.append(BitVec("retval_" + str(instr['address']), 256))
+        transaction = MessageCallTransaction(global_state.world_state,
+                                             environment.active_account,
+                                             environment.sender,
+                                             call_data,
+                                             environment.gasprice,
+                                             environment.callvalue,
+                                             environment.origin,
+                                             call_data_type,
+                                             callee_account.code
+                                             )
+        raise TransactionStartSignal(transaction, self.op_code)
 
-        environment = deepcopy(environment)
 
-        environment.code = callee_account.code
-        environment.calldata = call_data
+    @instruction
+    def delegatecall_post(self, global_state):
+        instr = global_state.get_current_instruction()
 
-        new_global_state = GlobalState(global_state.accounts, environment, global_state.node, MachineState(gas))
-        new_global_state.call_stack.append(global_state)
-        new_global_state.mstate.pc = -1
-        new_global_state.mstate.depth = global_state.mstate.depth + 1
-        new_global_state.mstate.constraints = copy(global_state.mstate.constraints)
+        try:
+            _, _, _, _, _, _, memory_out_offset, memory_out_size =\
+                get_call_parameters(global_state, self.dynamic_loader)
+        except ValueError as e:
+            logging.info(
+                "Could not determine required parameters for call, putting fresh symbol on the stack. \n{}".format(e)
+            )
+            global_state.mstate.stack.append(BitVec("retval_" + str(instr['address']), 256))
+            return [global_state]
 
-        return [new_global_state]
+        if global_state.last_return_data is None:
+            # Put return value on stack
+            return_value = BitVec("retval_" + str(instr['address']), 256)
+            global_state.mstate.stack.append(return_value)
+            global_state.mstate.constraints.append(return_value == 0)
+
+            return [global_state]
 
+            # Copy memory
+        global_state.mstate.mem_extend(memory_out_offset,
```

Comment:
It's possible for memory_out_offset to be some bitvec stuff right?

Reply:
fixed